File: app/src/main/java/com/example/test/CardEntityConverter.java
package com.example.test;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import javax.inject.Inject;

public class CardEntityConverter {

    private FilterCard filterCard;

    @Inject
    public CardEntityConverter(FilterCard filterCard) {
        this.filterCard = filterCard;
    }

    public CardEntity convertForHistory (Card card, String userBin, String timestamp) {
        return new CardEntity(userBin, card.getScheme(), card.getType(), card.getBank().getName(), timestamp);
    }

    public Map<String, String> convert (List<CardEntity> cardEntity) {
        Map<String, String> fromHistory = new LinkedHashMap<>();
        for (CardEntity entity : cardEntity) {
            fromHistory.put(entity.getBin(),
                    bankName(entity.getScheme()) + bankName(entity.getType()) +
                            bankName(entity.getBankName()) + entity.getTimestamp());
        }
        return filterCard.filterMap(fromHistory);
    }


    private String bankName(String string) {
        String value = "";
        if (string != null && !string.isEmpty()) {
            value = string + ", ";
        }
        return value;
    }
}

File: app/src/main/java/com/example/test/data/database/CardEntityConverter.kt
package com.example.test.data.database

import com.example.test.data.model.CardDto
import com.example.test.data.model.CardEntity
import javax.inject.Inject

class CardEntityConverter @Inject constructor(private var filterCard: FilterCard) {

    fun convertForHistory(card: CardDto, userBin: String, timestamp: String): CardEntity {
        return CardEntity(userBin, card.scheme, card.type, card.bank.name, timestamp)
    }

    fun convert(cardEntity: List<CardEntity>): Map<String, String> {
        val fromHistory = LinkedHashMap<String, String>()
        for (entity in cardEntity) {
            fromHistory[entity.bin] = bankName(entity.scheme) + bankName(entity.type) +
                bankName(entity.bankName) + entity.timestamp
        }
        return filterCard.filterMap(fromHistory);
    }

    private fun bankName(string: String?): String {
        return if (!string.isNullOrEmpty()) {
            "$string, "
        } else {
            ""
        }
    }
}